<?xml version="1.0"?>
<launch>

  <!-- Transforms -->

  <node pkg="tf2_ros" type="static_transform_publisher" name="gps_broadcaster" args="1 0 0 0 0 0 1 wamv/base_link gps" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="imu_broadcaster" args="1 0 0 0 0 0 1 wamv/base_link base_imu_link" />

  <!-- EKF node -->
  <node ns="wamv/robot_localization" pkg="robot_localization" type="ekf_localization_node" name="ekf_localization" clear_params="false">
	  <rosparam command="load" file="$(find gazebo_to_moos)/config/gps_imu_localization.yaml" />
  </node>

  <!-- Navsat node -->
  <node ns="wamv/robot_localization" pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" respawn="true" output="screen">
    <param name="magnetic_declination_radians" value="0"/>
    <param name="yaw_offset" value="0"/>
    <param name="zero_altitude" value="true"/>

    <param name="use_odometry_yaw" value="false"/>
    <param name="wait_for_datum" value="false"/>
   
    <param name="publish_filtered_gps" value="true"/>
    <param name="broadcast_utm_transform" value="false"/>
    
    <remap from="/imu/data" to="/imu" />
    <remap from="/gps/fix" to="/fix" />
    <remap from="/odometry/filtered" to="/wamv/robot_localization/odometry/filtered" />       
    <!-- rosparam param="datum">[24.173425, -120.692385, 0.0, wamv/odom, wamv/base_link]</rosparam -->
  </node>

  <node pkg="moos-ros-bridge" type="Bridge" name="Bridge"
        args="$(find gazebo_to_moos)/config/test.xml $(find gazebo_to_moos)/config/test.moos" 
        respawn="false" output="screen" />

  <node pkg="gazebo_to_moos" type="GazeboToMoos" name="GazeboToMoos" respawn="false" output="screen" >
     <rosparam command="load" file="$(find gazebo_to_moos)/config/GazeboToMoos.yaml" />
  </node>

</launch>
